### PR TITLE
Convert constant divides into multiplies

### PR DESCRIPTION
**Pull request by Hwurzburg**
I did not convert /2 or constant on constant since they  would not be run time divides (I think compiler is smart enough to convert /2 to shift instead of divide so did not change those)...

I am a bit confused as to when to use the "f"ending on constants....

This covers all of ArduPlane plus a few libraries not covered previously

btw.... #20289 made me do it!

### CODE REVIEW

File: ArduPlane/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -1213,7 +1213,7 @@ void GCS_MAVLINK_Plane::handleMessage(const mavlink_message_t &msg)
         Location new_home_loc {};
         new_home_loc.lat = packet.latitude;
         new_home_loc.lng = packet.longitude;
-        new_home_loc.alt = packet.altitude / 10;
+        new_home_loc.alt = packet.altitude * 0.1f;
```
Comment (WickedShell) [inline ArduPlane/GCS_Mavlink.cpp:1216]: This is wrong, it was an integer before, this is forcing a promotion to floating point type, then casting back down to an integer.

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -1373,7 +1373,7 @@ uint8_t GCS_MAVLINK_Plane::high_latency_tgt_heading() const
 #endif
         const AP_Navigation *nav_controller = plane.nav_controller;
         // need to convert -18000->18000 to 0->360/2
-        return wrap_360_cd(nav_controller->target_bearing_cd() ) / 200;
+        return wrap_360_cd(nav_controller->target_bearing_cd() ) * 0.05f;
```
Comment (WickedShell) [inline ArduPlane/GCS_Mavlink.cpp:1376]: Same thing this is forcing something to a float that was not before.

**Review by WickedShell — CHANGES_REQUESTED** [review]
A lot of these look incorrect, you are forcing things from integer types into floating point types then back to integer types at the end. I stopped commenting on them after the first couple, but a lot of these need to be redone, because they aren't improvements.

File: libraries/AP_RSSI/AP_RSSI.cpp [hunk 3]
```diff
@@ -141,7 +141,7 @@ float AP_RSSI::read_receiver_rssi()
         case RssiType::RECEIVER: {
             int16_t rssi = RC_Channels::get_receiver_rssi();
             if (rssi != -1) {
-                return rssi / 255.0;
+                return rssi * 0.00392; // divide by 255
```
Comment (WickedShell) [inline libraries/AP_RSSI/AP_RSSI.cpp:144]:
```suggestion
                return rssi * (1/255f);
```

Same result, but it's much clearer where the constant came from, and will compile to the same thing

File: ArduPlane/tiltrotor.cpp [hunk 4]
```diff
@@ -516,9 +516,9 @@ void Tiltrotor::vectoring(void)
                 // base the tilt on elevon mixing, which means it
                 // takes account of the MIXING_GAIN. The rear tilt is
                 // based on elevator
-                const float right = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_right) / 4500.0;
-                const float left  = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_left) / 4500.0;
-                const float mid  = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevator) / 4500.0;
+                const float right = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_right) * 0.0002222; // 0.0002222 = 1/4500
```
Comment (WickedShell) [inline ArduPlane/tiltrotor.cpp:519]: Just include the constant you want as an actual floating point calculation. This is very hard to read, and arbitrarily limits accuracy of the result due to truncation of the provided constant.

File: ArduPlane/mode_loiter.cpp [hunk 5]
```diff
@@ -58,7 +58,7 @@ bool ModeLoiter::isHeadingLinedUp_cd(const int32_t bearing_cd)
     */
 
     // Use integer division to get discrete steps
-    const int32_t expanded_acceptance = 1000 * (labs(plane.loiter.sum_cd) / 36000);
+    const int32_t expanded_acceptance = 1000 * (labs(plane.loiter.sum_cd) * 0.00002777f; // 0.00002777 = divide by 36,000
```
Comment (WickedShell) [inline ArduPlane/mode_loiter.cpp:61]: Floating point problems, plus out of scope with the above comment, needs a bit more looking at the usage and less mechanical replacement.

**Comment by Hwurzburg** [comment]
reverted anything that forced floats and adopted suggestions for how to show fractions instead of single constant...sorry still learning

File: ArduPlane/tiltrotor.cpp [hunk 6]
```diff
@@ -516,9 +516,9 @@ void Tiltrotor::vectoring(void)
                 // base the tilt on elevon mixing, which means it
                 // takes account of the MIXING_GAIN. The rear tilt is
                 // based on elevator
-                const float right = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_right) / 4500.0;
-                const float left  = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_left) / 4500.0;
-                const float mid  = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevator) / 4500.0;
+                const float right = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_right) * (1/4500.0);
```
Comment (IamPete1) [inline ArduPlane/tiltrotor.cpp:519]: we have a `SERVO_MAX` define for the 4500 somewhere. But whatever.

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/tiltrotor.cpp [hunk 7]
```diff
@@ -305,8 +305,8 @@ void Tiltrotor::continuous_update(void)
         // Q_TILT_MAX. Anything above 50% throttle gets
         // Q_TILT_MAX. Below 50% throttle we decrease linearly. This
         // relies heavily on Q_VFWD_GAIN being set appropriately.
-       float settilt = constrain_float((SRV_Channels::get_output_scaled(SRV_Channel::k_throttle)-MAX(plane.aparm.throttle_min.get(),0)) / 50.0f, 0, 1);
-       slew(MIN(settilt * max_angle_deg / 90.0f, get_forward_flight_tilt()));
+       float settilt = constrain_float((SRV_Channels::get_output_scaled(SRV_Channel::k_throttle)-MAX(plane.aparm.throttle_min.get(),0)) / 0.02, 0, 1);
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply ArduPlane/tiltrotor.cpp]: fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**